cis194/week1/jimfang/Homework01.hs

```diff
@@ -0,0 +1,34 @@
+module Homework01 (
+    toDigits,
+    toDigitsRev,
+    doubleEveryOther,
+    sumDigits,
+    validate
+) where
+
+-- toDigits
+toDigits :: Integer -> [Integer]
+toDigits num
+  | num <= 0 = []
+  | otherwise = toDigits ( num `div` 10 ) ++ [ num `mod` 10 ]
+
+-- toDigitRev
+toDigitsRev :: Integer -> [Integer]
+toDigitRev = reverse . toDigits
+
+-- doubleEveryOther
+doubleEveryOther :: [Integer] -> [Integer]
+doubleEveryOther []         = []
+doubleEveryOther [x]        = [x]
+doubleEveryOther [x:y:rest] = [x: y*2] ++ doubleEveryOther rest
+
+-- sumDigits
+sumDigits :: [Integer] -> Integer
+sumDigits = sum . concatMap . toDigits
+
+-- validate
+tenTimes :: Integer -> Bool
+tenTimes num = num `mod` 10 == 0
+
+validate :: Integer -> Bool
+validate = tenTimes . sumDigits . doubleEveryOther . toDigits 
```

Comment:
I think the requirement is doubling every other digit from the right. Is this logic correct?